node/src/components/consensus/era_supervisor.rs

```diff
@@ -865,6 +897,29 @@ where
         self.handle_consensus_results(self.era_supervisor.current_era, results)
     }
 
+    /// Handles registering a new upgrade activation point.
+    pub(super) fn got_upgrade_activation_point(
+        &mut self,
+        maybe_activation_point: Option<ActivationPoint>,
+    ) -> Effects<Event<I>> {
+        match (
+            maybe_activation_point,
+            &self.era_supervisor.next_upgrade_activation_point,
+        ) {
+            (Some(new_point), Some(current_point)) => {
+                if new_point != *current_point {
+                    warn!(%new_point, %current_point, "unexpected mismatch in activation points");
+                }
+            }
```

Comment:
Could that happen if we had planned an upgrade for some era, but then we found a pressing issue and decided to push another upgrade earlier?

Reply:
Hmm - I hadn't anticipated a need for that, but it's a good point.

At the moment this code path shouldn't hit, as we stop polling for new activation points once we learn of a new one.

I'll update the code to accommodate your idea.

Reply:
Updated in [4e4722a](https://github.com/CasperLabs/casper-node/pull/843/commits/4e4722a9eb42fcdab0931c2dd7e8311871be44e0).